### ISSUE TITLE
NotImplementedError: Unable to parse schema object '{'required': ['runGraphIndex']}' with no type and no reference

### ISSUE THREAD

**Issue by sampart** [GrammaTech/pylint-sarif#1]
When running `pylint2sarif.py` under either Python 2 or 3, I see the following error and no SARIF is generated:

```
NotImplementedError: Unable to parse schema object '{'required': ['runGraphIndex']}' with no type and no reference
```

Here's the full traceback I'm given:
<details>

```
Traceback (most recent call last):
  File "../pylint-sarif/pylint2sarif.py", line 328, in <module>
    main()
  File "../pylint-sarif/pylint2sarif.py", line 41, in main
    p2s = Pylint2Sarif(args)
  File "../pylint-sarif/pylint2sarif.py", line 137, in __init__
    self.sarif = builder.build_classes()
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/__init__.py", line 136, in build_classes
    builder.construct(uri, defn, **kw)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 497, in construct
    ret = self._construct(uri, *args, **kw)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 586, in _construct
    self.resolved[uri] = self._build_object(uri, clsdata, parent, **kw)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 683, in _build_object
    typ = self.resolve_type(detail["items"]["$ref"], nm)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 486, in resolve_type
    self.resolved[uri] = self.construct(uri, resolved, (ProtocolBase,))
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 497, in construct
    ret = self._construct(uri, *args, **kw)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 586, in _construct
    self.resolved[uri] = self._build_object(uri, clsdata, parent, **kw)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 683, in _build_object
    typ = self.resolve_type(detail["items"]["$ref"], nm)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 486, in resolve_type
    self.resolved[uri] = self.construct(uri, resolved, (ProtocolBase,))
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 497, in construct
    ret = self._construct(uri, *args, **kw)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 511, in _construct
    klasses = self.construct_objects(clsdata["oneOf"], uri)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 778, in construct_objects
    for i, item_detail in enumerate(oneOfList)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 778, in <listcomp>
    for i, item_detail in enumerate(oneOfList)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 497, in construct
    ret = self._construct(uri, *args, **kw)
  File "/Users/sam/code/other/kashflow-python-samples/venv/lib/python3.7/site-packages/python_jsonschema_objects/classbuilder.py", line 602, in _construct
    "no type and no reference".format(clsdata)
NotImplementedError: Unable to parse schema object '{'required': ['runGraphIndex']}' with no type and no reference
```

Although this is running within a virtualenv, I've also tried this with a global install and got the same error.
</details>

Versions:
<details>

I'm on MacOS.  My Python versions are Python 3.7.6 / Python 2.7.16

`pylint --version` gives me:

```
pylint 2.4.4
astroid 2.3.3
Python 3.7.6 (default, Dec 30 2019, 19:38:26) 
[Clang 11.0.0 (clang-1100.0.33.16)]
```

Looking inside `python_jsonschema_objects/_version.py`, I see

```
version_json = '''
{
 "dirty": false,
 "error": null,
 "full-revisionid": "3f3cb9090c34016499b183cb14ca01d3cce4d46f",
 "version": "0.3.12"
}
```
</details>

The little I could find online suggests that the error I'm seeing comes from `python_jsonschema_objects`, but I couldn't get any further than that.

**Comment by fishoak** [comment]
Thanks for this report. I'm a bit swamped right now, but hope to get to look at it in a few days.

**Comment by fishoak** [comment]
This was fixed in 79bb7e1c6d6cfc7f22c2e5e40595fbeddbf9a6e0.

**Comment by sampart** [comment]
Thank you!
